include/kos/time.h

```diff
@@ -35,48 +34,94 @@
 
 __BEGIN_DECLS
 
-#if !defined(__STRICT_ANSI__) || (__STDC_VERSION__ >= 201112L) || (__cplusplus >= 201703L)
+/** \cond */
 
-/* Forward declaration. */
-struct timespec;
+/* Required definition for a fully C23-compliant <time.h> */
+#define __STDC_VERSION_TIME_H__  202311L
 
-#define TIME_UTC 1
+/* Microsecond resolution for clock(), per POSIX standard.. */
+#define CLOCKS_PER_SEC           1000000
 
-/* Microsecond resolution for clock(), per POSIX standard */
-#define CLOCKS_PER_SEC 1000000
+/* =============== Enable the following for >=c11, >=c++17 =================== */
+#if (defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L)) || \
+    (defined(__cplusplus)      && (__cplusplus >= 201703L))
 
-/* C11 nanosecond-resolution timing. */
-extern int timespec_get(struct timespec *ts, int base);
+   /* Only supported base time in C11. */
+   #define TIME_UTC 1
```

Comment:
Please don't indent things in like this inside #if blocks. We don't do that anywhere else in the code, so please don't do it here. And **especially** don't indent it by 3 spaces like this.

Note: this applies to the rest of the file too, of course.